### PR TITLE
Quiz : "Questions ordered" option using categories doesn't work properly (questions are not ordered)

### PR DESCRIPTION
**Pull request by carlangas159**
Minor - Fix table on settings to select max for category's
image

EX_Q_SELECTION_CATEGORIES_RANDOM_QUESTIONS_ORDERED = 4
Questions are arranged alphabetically based on the text of the question.

EX_Q_SELECTION_CATEGORIES_ORDERED_QUESTIONS_RANDOM = 5
Questions are arranged alphabetically based on the category and with the question field.

#3265

### CODE REVIEW

File: main/exercise/exercise.class.php [hunk 1]
```diff
@@ -10569,4 +10585,46 @@ private function setResultDisabledGroup(FormValidator $form)
...
+            $order = 'DESC';
+        }
+
+        $sql = "SELECT
+                   iid
+                FROM
+                    $tableQuiz
+                WHERE
+                      iid in ($jsonQuestionId)
+                ORDER BY question $order;";
```
Comment (ywarnier) [inline main/exercise/exercise.class.php:10622]: This ; before the " is not necessary.

File: main/exercise/exercise.class.php [hunk 2]
```diff
@@ -10569,4 +10585,46 @@ private function setResultDisabledGroup(FormValidator $form)
...
+    public function getAlfabeticalOrderQuestion(
+        $questionsId = [],
+        $orderAsc = true
+    ) {
+        if (empty($questionsId)) return $questionsId;
+        if (!is_array($questionsId)) return $questionsId;
+
+        $tableQuiz = Database::get_course_table(TABLE_QUIZ_QUESTION);
+        $jsonQuestionId = json_encode($questionsId);
+        $jsonQuestionId = str_replace(['[', ']', '"', "'"], '', $jsonQuestionId);
```
Comment (ywarnier) [inline main/exercise/exercise.class.php:10609]: If the goal is only to split a flat (one dimension) array with commas, then you should do something easier, like 
```
implode(',', $questionsId);
```
The current version seems less efficient.

File: main/exercise/exercise.class.php [hunk 3]
```diff
@@ -10569,4 +10585,45 @@ private function setResultDisabledGroup(FormValidator $form)
...
+     */
+    public function getAlfabeticalOrderQuestion(
+        $questionsId = [],
+        $orderAsc = true
+    ) {
+        if (empty($questionsId)) return $questionsId;
+        if (!is_array($questionsId)) return $questionsId;
+
+        $tableQuiz = Database::get_course_table(TABLE_QUIZ_QUESTION);
+        $QuestionsId = implode(',', $questionsId);
```
Comment (ywarnier) [inline main/exercise/exercise.class.php:10608]: $QuestionsId should be $questionsId or something else without uppercase Q

File: main/exercise/exercise.class.php [hunk 4]
```diff
@@ -10210,6 +10222,10 @@ private function pickQuestionsPerCategory(
                     );
 
                     if (!empty($elements)) {
+                        $questionSelectionType = $this->getQuestionSelectionType();
+                        if($questionSelectionType == EX_Q_SELECTION_CATEGORIES_ORDERED_QUESTIONS_ORDERED){
```
Comment (ywarnier) [inline main/exercise/exercise.class.php:10226]: Missing one space before and after parenthesis

File: main/exercise/TestCategory.php [hunk 5]
```diff
@@ -535,7 +537,10 @@ public static function getQuestionsByCat(
                     exercice_id = $exerciseId AND
                     qrc.c_id = $courseId
                 ";
-
+        // order alphabetical by category and question  = 3
+        if($questionSelectionType == EX_Q_SELECTION_CATEGORIES_ORDERED_QUESTIONS_ORDERED){
```
Comment (ywarnier) [inline main/exercise/TestCategory.php:541]: Missing spaces before/after ()

File: main/exercise/TestCategory.php [hunk 6]
```diff
@@ -535,7 +537,10 @@ public static function getQuestionsByCat(
                     exercice_id = $exerciseId AND
                     qrc.c_id = $courseId
                 ";
-
+        // order alphabetical by category and question  = 3
+        if ($questionSelectionType == EX_Q_SELECTION_CATEGORIES_ORDERED_QUESTIONS_ORDERED) {
+            $sql .= " ORDER BY q.question ASC ";
```
Comment (AngelFQC) [inline main/exercise/TestCategory.php:542]: Las preguntas no se ordenan alfabéticamente.

**Comment by AngelFQC** [comment]
Este PR no va, pero el bug sigue. He logrado reproducir el error en el orden de preguntas.

**Pull request closed without merging**